### PR TITLE
Use Locale.ROOT for toLowerCase/toUpperCase operations

### LINKED ISSUE #1414: Spock does not work in tr_TR locale: Unrecognized block label: given @ line 31, column 13.
### Describe the bug

Spock fails with `tr_TR` locale:

```
D:\a\jmeter\jmeter\src\core\src\test\groovy\org\apache\jmeter\engine\util\FunctionParserSpec.groovy: 31: Unrecognized block label: given @ line 31, column 13.
               CompoundVariable.functions.put('__func', Func.class)
               ^
```

It blocks JMeter testing :-/
https://github.com/apache/jmeter/pull/693

Reason:
Spock uses `.toLowerCase()` for `BlockParseInfo`:
https://github.com/spockframework/spock/blob/3ebb5d38f1a342935c97f4c331276ae4bcf515f6/spock-core/src/main/java/org/spockframework/compiler/model/BlockParseInfo.java#L139-L141

The lookup fails here:
https://github.com/spockframework/spock/blob/3ebb5d38f1a342935c97f4c331276ae4bcf515f6/spock-core/src/main/java/org/spockframework/compiler/SpecParser.java#L212-L225

Sample code in question (I believe it does not matter):

```groovy
import org.apache.jmeter.functions.Function
import org.apache.jmeter.samplers.SampleResult
import org.apache.jmeter.samplers.Sampler

import spock.lang.Specification
import spock.lang.Unroll

@Unroll
class FunctionParserSpec extends Specification {
    def "function '#value' gets compiled"() {
        given:
            CompoundVariable.functions.put('__func', Func.class)
            def parser = new FunctionParser()
        when:
            def result = parser.compileString(value)
        then:
            "$result" == "$expected"
        where:
            value           | expected
            '${__func()}'   | [new Func()]
            '${ __func()}'  | [new Func()]
            '${__func() }'  | [new Func()]
            '${ __func() }' | [new Func()]
    }
```

### To Reproduce

run any Spock test with `-Duser.country=TR -Duser.language=tr`

### Expected behavior

tr_TR should be supported

### Actual behavior

Unrecognized block label: given @ line 31, column 13

### Java version

8, 11, 17

### Buildtool version

Gradle 7.3

### What operating system are you using

Windows

### Dependencies

I believe it is irrelevant

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leonard84**
Prior to this commit, there were several instances of toLowerCase()/toUpperCase()
using the default locale, this causes issues in certain locales like tr_TR.
To fis this, we declare Locale.ROOT should be used in those cases.

fixes #1414

### CODE REVIEW

File: spock-core/src/main/java/org/spockframework/compiler/model/BlockParseInfo.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package org.spockframework.compiler.model;
 
-import java.util.EnumSet;
+import java.util.*;
```
Comment (szpak) [inline spock-core/src/main/java/org/spockframework/compiler/model/BlockParseInfo.java:19]: Why this change for just one import?

File: spock-spring/src/main/java/org/spockframework/spring/xml/SpockMockFactoryBean.java [hunk 2]
```diff
@@ -19,7 +19,7 @@
 import org.spockframework.mock.MockNature;
 import spock.mock.DetachedMockFactory;
 
-import java.util.Collections;
+import java.util.*;
```
Comment (szpak) [inline spock-spring/src/main/java/org/spockframework/spring/xml/SpockMockFactoryBean.java:22]: Similarly.

**Review by szpak — APPROVED** [review]
Some nitpicking.

File: spock-core/src/main/java/org/spockframework/compiler/model/BlockParseInfo.java (hunk 1 above)
Comment (leonard84) [reply spock-core/src/main/java/org/spockframework/compiler/model/BlockParseInfo.java:19]: It's not just one import `java.util.Locale` was added.

**Pull request merged**